lib/orocos/process.rb

```diff
@@ -240,6 +247,14 @@ def register_task(task)
             @tasks << task
         end
 
+        def register_ior(task_name, ior)
+            @task_ior_mappings[task_name] = ior
+        end
+
+        def has_ior_for?(task_name)
+            @task_ior_mappings.fetch(task_name, nil)
```

Comment:
.key?(...)

Reply:
I used this method to get the IOR in case it was registered. Maybe I should rename it to `ior_for`?

Reply:
I think he means to use `.key?` instead of `.fetch`

Reply:
Yeah, I got that. My point was that I was using this method to fetch the IOR or return nil if it is not there, not only to know whether the task_name is a key on the hash.

Reply:
Then do not call the method `has_ior_for?` which really, explicitly, should only be used in a boolean context.